kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/KafkaClient.java

```diff
@@ -30,7 +32,20 @@ public interface KafkaClient {
      * @param message the message
      * @param numOfMessages the num of messages
      */
-    void produceMessages(String topicName, String bootstrap, String message, int numOfMessages) throws KubeClusterException;
+    default void produceMessages(String topicName, String bootstrap, String message, int numOfMessages) throws KubeClusterException {
+        produceMessages(topicName, bootstrap, message, null, numOfMessages);
+    }
+
+    /**
+     * Produce messages.
+     *
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param message the message
+     * @param messageKey optional record key for the message. <code>null</code> means don't specify a key
+     * @param numOfMessages the num of messages
+     */
+    void produceMessages(String topicName, String bootstrap, String message, @Nullable String messageKey, int numOfMessages) throws KubeClusterException;
```

Comment:
This API choice is in keeping with specifying a single `message` argument. However that limits things to a single key for all messages in the call, which could be acceptable. 

However I wonder if something along the lines of 
```suggestion
public void produceMessages(String topicName, String bootstrap, String message, @Nullable Function<Integer, String> messageKeyGenerator, int numOfMessages)
```
So that callers could supply something like 
```java
(messageNumber ->  "key_" + messageNumber)
```

Might prove more flexible

Reply:
I'd be happy with your idea.  I'd been toying with the idea of this being a caller supplying a Stream<ProducerRecord>.  That would give the caller the opportunity to vary the record, and let them supply a key if they wished.  That would eliminate the numOfMessages argument.

Reply:
edit: maybe something ProducerRecord-like rather than `ProducerRecord` itself.  That avoids needing to deal with `ProducerRecord#topicName` and `ProducerRecord#partition` as non of the CLIs allows that degree of control on a per message basis.

Reply:
I quite like the stream idea as well but I don't think it plays very will with CLI clients (certainly not with out a lot of refactoring so we can consume the records incrementally). 

I was experimenting with just
```java
record KafkaMessage(@Nullable String recordKey, String value);
```

The other thing is the number of messages arg seems to be quite well understood by the CLI tools so while I'm not a huge fan of the arg it does feel like a reasonable API.

Reply:
Should I merge what I have now or should I try the generator function?

Reply:
Happy for it to be merged as it stands.

Reply:
Still happy for it to be merged as it stands.